chatbot-core/rag/embedding/embedding_utils.py

```diff
@@ -0,0 +1,30 @@
+"""
+Utility functions for loading a sentence transformer model and embedding text documents.
+"""
+
+from sentence_transformers import SentenceTransformer
+
+def load_embedding_model(model_name, logger):
+    """
+    Load the sentence transformer model for generating text embeddings.
+
+    Returns:
+        SentenceTransformer: The loaded embedding model.
```

Comment:
check if model is a SentenceTransformer instance. `if not isinstance(model, SentenceTransformer):
    raise TypeError("Model must be a SentenceTransformer instance.")
`